# 📊 Descriptive Statistics on Iris Dataset

## Objective
To compute statistical measures such as mean, median, standard deviation, and percentiles for different species in the Iris dataset and analyze their differences.

In [14]:
import pandas as pd
import numpy as np

In [15]:
df = pd.read_csv("Iris.csv")
df.head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,1,5.1,3.5,1.4,0.2,Iris-setosa
1,2,4.9,3.0,1.4,0.2,Iris-setosa
2,3,4.7,3.2,1.3,0.2,Iris-setosa
3,4,4.6,3.1,1.5,0.2,Iris-setosa
4,5,5.0,3.6,1.4,0.2,Iris-setosa


The dataset contains measurements of iris flowers including sepal length, sepal width, petal length, petal width, and species.

In [16]:
df.shape

(150, 6)

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


In [18]:
df.describe()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm
count,150.000000,150.000000,150.000000,150.000000,150.000000
mean,75.500000,5.843333,3.054000,3.758667,1.198667
std,43.445368,0.828066,0.433594,1.764420,0.763161
min,1.000000,4.300000,2.000000,1.000000,0.100000
25%,38.250000,5.100000,2.800000,1.600000,0.300000
50%,75.500000,5.800000,3.000000,4.350000,1.300000
75%,112.750000,6.400000,3.300000,5.100000,1.800000
max,150.000000,7.900000,4.400000,6.900000,2.500000


### 🔹 Observation
- Dataset contains 150 samples and 5 columns  
- Includes 3 species: Iris-setosa, Iris-versicolor, Iris-virginica  

In [19]:
grouped = df.groupby('Species')

In [20]:
grouped.mean()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm
Species,,,,,
Iris-setosa,25.5,5.006,3.418,1.464,0.244
Iris-versicolor,75.5,5.936,2.770,4.260,1.326
Iris-virginica,125.5,6.588,2.974,5.552,2.026


In [21]:
grouped.median()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm
Species,,,,,
Iris-setosa,25.5,5.0,3.4,1.50,0.2
Iris-versicolor,75.5,5.9,2.8,4.35,1.3
Iris-virginica,125.5,6.5,3.0,5.55,2.0


In [22]:
grouped.std()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm
Species,,,,,
Iris-setosa,14.57738,0.352490,0.381024,0.173511,0.107210
Iris-versicolor,14.57738,0.516171,0.313798,0.469911,0.197753
Iris-virginica,14.57738,0.635880,0.322497,0.551895,0.274650


In [23]:
grouped.describe()

Id                                                        \
                count   mean       std    min     25%    50%     75%    max   
Species                                                                       
Iris-setosa      50.0   25.5  14.57738    1.0   13.25   25.5   37.75   50.0   
Iris-versicolor  50.0   75.5  14.57738   51.0   63.25   75.5   87.75  100.0   
Iris-virginica   50.0  125.5  14.57738  101.0  113.25  125.5  137.75  150.0   

                SepalLengthCm         ... PetalLengthCm      PetalWidthCm  \
                        count   mean  ...           75%  max        count   
Species                               ...                                   
Iris-setosa              50.0  5.006  ...         1.575  1.9         50.0   
Iris-versicolor          50.0  5.936  ...         4.600  5.1         50.0   
Iris-virginica           50.0  6.588  ...         5.875  6.9         50.0   

                                                           
                  mean       std  min  25%  50%  75%  max  
Species                                                    
Iris-setosa      0.244  0.107210  0.1  0.2  0.2  0.3  0.6  
Iris-versicolor  1.326  0.197753  1.0  1.2  1.3  1.5  1.8  
Iris-virginica   2.026  0.274650  1.4  1.8  2.0  2.3  2.5  

[3 rows x 40 columns]

In [24]:
np.percentile(df['SepalLengthCm'], [25, 50, 75])

array([5.1, 5.8, 6.4])

In [25]:
setosa_list = df[df['Species'] == 'Iris-setosa']['SepalLengthCm'].tolist()
versicolor_list = df[df['Species'] == 'Iris-versicolor']['SepalLengthCm'].tolist()
virginica_list = df[df['Species'] == 'Iris-virginica']['SepalLengthCm'].tolist()

print("Setosa:", setosa_list[:5])
print("Versicolor:", versicolor_list[:5])
print("Virginica:", virginica_list[:5])

Setosa: [5.1, 4.9, 4.7, 4.6, 5.0]
Versicolor: [7.0, 6.4, 6.9, 5.5, 6.5]
Virginica: [6.3, 5.8, 7.1, 6.3, 6.5]


### 🔹 Observations

- Iris-setosa has smaller petal length and width compared to other species  
- Iris-virginica generally has the largest measurements  
- Iris-versicolor lies between setosa and virginica  
- Mean and median values help understand central tendency  
- Standard deviation shows variability within each species  

## ✅ Conclusion

Descriptive statistics help summarize and understand the distribution of data.  
Grouping by species allows comparison between different categories effectively.